src/analysis/retail/evoker/preservation/modules/talents/CycleOfLife.tsx

```diff
@@ -30,30 +35,40 @@ const TRACKED_SPELL_IDS = [
 
 class CycleOfLife extends Analyzer {
   savedBySpell: Map<number, number> = new Map<number, number>();
+  seeds: number[] = []; // storing seed expiration times
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.CYCLE_OF_LIFE_TALENT);
     this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.handleHeal);
+    this.addEventListener(
+      Events.summon.by(SELECTED_PLAYER).spell(SPELLS.CYCLE_OF_LIFE_SUMMON),
+      this.onSummon,
+    );
+  }
+
+  onSummon(event: SummonEvent) {
+    this.seeds.push(event.timestamp + CYCLE_OF_LIFE_SEED_DURATION + 150); // buffer for lag
   }
 
   handleHeal(event: HealEvent) {
-    if (
-      !this.selectedCombatant.hasBuff(SPELLS.CYCLE_OF_LIFE_BUFF.id) &&
-      !this.selectedCombatant.hasBuff(SPELLS.CYCLE_OF_LIFE_STACK.id)
-    ) {
-      return;
-    }
     const curVal = this.savedBySpell.has(event.ability.guid)
       ? this.savedBySpell.get(event.ability.guid)
       : 0;
     const amount = event.amount + (event.absorbed || 0) + (event.overheal || 0);
-    const percentSaved = this.selectedCombatant.hasBuff(SPELLS.CYCLE_OF_LIFE_BUFF.id)
-      ? CYCLE_OF_LIFE_PERCENT_SAVED
-      : CYCLE_OF_LIFE_PERCENT_SAVED *
-        this.selectedCombatant.getBuffStacks(SPELLS.CYCLE_OF_LIFE_STACK.id);
+    const percentSaved = CYCLE_OF_LIFE_PERCENT_SAVED * this.getNumActiveSeeds(event.timestamp);
     this.savedBySpell.set(event.ability.guid, curVal! + amount * percentSaved);
   }
 
+  getNumActiveSeeds(timestamp: number): number {
```

Comment:
This is probably fine but why not use the cycle's heal event as your trigger to prune expired seeds?

Reply:
You can do that but it's gonna be less efficient because you'll have another event listener. Also, you'll have to make sure you only prune it once for each cycle of life group and then it just adds unneeded complexity imo.